### PR TITLE
feat: allow double equal checks

### PR DESCRIPTION
**Pull request by anonrig**
Double equal is used to simplify `value === undefined && value === null`. We shouldn't disallow them.

Ref: https://github.com/getsentry/sentry/pull/61593

### CODE REVIEW

**Comment by ryan953** [comment]
@anonrig Can we expand a little more on what the problem is here, who's impact and why we're making a change?

**Comment by anonrig** [comment]
> @anonrig Can we expand a little more on what the problem is here, who's impact and why we're making a change?

While replacing certain lodash functions with native ones, I replaced lodash/isNil with directly correlates to "value == null" which checks for both undefined and null values. If disallowed we should have checked with "value === null && value === undefined" which makes the call longer and more redundant. 

I don't see any point for disallowing double equal operator. 

For example, a large project like Node.js uses double equal operator in its codebase and it is not "disallowed" by the linter.

**Comment by billyvg** [comment]
I think it's more clear/obvious when being explicit with `value === null && value === undefined` rather than assuming the reader knows that `undefined == null`. `isNil` doesn't seem to be used that often where using the longer form will be frustrating

**Comment by evanpurkhiser** [comment]
> The same can be said for "??" operator

I think there's a pretty huge difference in an operator like nullish coalescing versus the double / triple equal operators in terms of ubiquity.

We have a large variety of engineers, many of whom may not work in JavaScript daily who are more familiar with at least one other language where `==` is a strict equality check. In JavaScript, it is not and this can definitely lead to real bugs. I do not think we should expect that we can teach an entire engineering organization this over-night. Most engineers reviewing code are not going to catch this subtle difference in a PR, and frankly, it's not something any one of us should have to think about in a review ("Wait should this be a double or triple equal").

Even if we did manage to teach our entire organization the difference in when to use which, we'll have to keep repeating that process for every new engineer who we onboard into the javascript project. The trade-off here of some convenience (and possibly negligible performance benefits) is not worth the continued effort of education and the additional burden on a reviewer to catch misusages.

Standardizing on just using a triple equal means we simply never have to think about this, at the very small cost of it being sometimes slightly more verbose.

**Review by ryan953 — CHANGES_REQUESTED** [review]
**TL/DR**: Requesting changes so this doesn't get accidentally merged

---

I strongly agree with the comments in the thread. To summarize what I'm seeing, these (paraphrased) statements made above all very true:
a) Changing this rule brings up an education problem, people can be taught the difference between null and undefined.
b) The convenience of `==` in a few spots is not that great, anyone can write `value === null && value === undefined` without any trouble
c) JS has some language features that are easy to misuse (the list of bad parts have changed since this book was written, but it's still there) and it's valuable to reduce the amount of parsing people need to do when they read code. People are, of course, the least efficient/consistent/correct parsers available.

I'm strongly against this rule change for points b and c above.

---

Even if there was consensus that we should change this rule, and there is not, The first step in the process is to merge this PR. The education piece is by far more important and difficult.

If we wanted to make this lint rule change, or a change to a similar lint rule, we must address the education aspect first and strongly. The first step should not be to open ourselves up to bugs in the code because someone didn't realize the change has happened.

There's no shortage of similar lint rules: could be anything from semi-colon requirements, to curly braces around one-line conditional blocks, or even using nullish coalescing. Amending any of these is really just off the table IMO.
